### ISSUE TITLE
S1121 flags record's with operation

### ISSUE THREAD

**Issue by iSeiryu** [SonarSource/sonar-dotnet#3998]
### Description

https://rules.sonarsource.com/csharp/RSPEC-1121 flags this code
```csharp
record Person(string FirstName);

var person = new Person("Steve");
var mike = person with { FirstName = "Mike" };
```

### Expected behavior

`FirstName = "Mike"` should not be flagged by this rule.

### Known workarounds
Disable the rule.

### Related information

* SonarAnalyzer.CSharp 8.17.0.26580
* Visual Studio 2019
* dotnet 5.0
* Windows 10

**Comment by pavel-mikula-sonarsource** [comment]
Hi @iSeiryu ,

Thank you for reporting this case, I can confirm it as a False Positive. We already have this case documented in your unit tests and we'll add support for C# 9 Records in [MMF-2223](https://jira.sonarsource.com/browse/MMF-2223) so I'll close this issue for now.
